src/Core/Grid/Definition/Factory/CustomerAddressGridDefinitionFactory.php

```diff
@@ -54,7 +54,24 @@ final class CustomerAddressGridDefinitionFactory extends AbstractGridDefinitionF
     public function __construct(HookDispatcherInterface $hookDispatcher, ?Request $currentRequest)
     {
         parent::__construct($hookDispatcher);
-        $this->backUrl = $currentRequest ? $currentRequest->getUri() : '';
+
+        // Remove useless query parameters to avoid some strange behaviors with delete workflow
```

Comment:
I think this comment is not accurate.

These query parameters are not useless, they are not wanted. The behavior we have today is not strange, it's not what we want.